app/models/loan.rb

```diff
@@ -64,6 +70,8 @@ class Loan < ActiveRecord::Base
   attr_option_settable :status, :loan_type, :project_type, :public_level
 
 
+  attr_custom_model_linkable LOAN_CRITERIA_FIELD_SET_SYMBOL, POST_ANALYSIS_FIELD_SET_SYMBOL
```

Comment:
I can understand that, I actually wasn't allowed to write tests at all at my last job, our deadlines were sometimes measured in days or weeks rather than on a scale of a month or more, but it's a way of thinking I came to believe was very very important doing my own work on my own side projects and a skill set that I've tried very hard to build up in spite of that. 

My concerns re: testing aren't purely ideological, but based on a foundation of seeing these practices result in much higher-quality and more maintainable code, and reinforced by the instability of the software produced under those environments with strong time pressures or institutional constraints that prevented good regression testing.

The ruby world has *much* better support for meaningful tests. When we start having enough UI to do them, most of the specs should end up being "feature" specs, which are integration tests that actually run a headless renderer and navigate the platform the same way an end-user does. The ability to quickly and reliably write tests that work the entire system in roughly the same way a user does makes them invaluable.

The scope of "unit tests" in rails work tends to be a little more narrow, which is part of the reason that I like to use shared example groups for functionality used in multiple models where their interface in the calling model is pretty simplistic. Mostly we just want to know "do the custom methods manually defined on this model behave the way they are supposed to?", we don't test basic rails functionality like scopes, saving records, or even validations for the most part. Finding where to make those tradeoffs is more art than science, and I understand the frustration (I was pulling my hair out the first, oh, 4 or 5 projects where I really tried to do rigorous testing like this), but I promise that I'm not asking you to make these sorts of changes for no reason :sweat_smile: .